In [1]:
import numpy as np
from IPython.display import HTML
from bokeh.plotting import output_notebook, show
import genomes_dnj.lct_interval.series_plots as dm
import genomes_dnj.lct_interval.anal_series as an
import genomes_dnj.lct_interval_snp_anal.lct_interval_snp_anal as snp_anal
import genomes_dnj.lct_interval.series_masks as sm
from genomes_dnj.lct_interval.bokeh_match_count_plot import snp_match_count_plot_cls as smcp
output_notebook(hide_banner=True)
sa_123_1561 = snp_anal.series_anal_cls(dm.di_123_1561, sm.series_data)

<h3>Series 123_1561</h3>
<div style="width:700px">
<p>
This notebook documents the series 123_1561 and its 123 SNPs. The 1000 genomes data includes 2948
chromosome 2 samples that express none of these 123 SNPs and 1186 samples that express all of them.
Another 282 express 122 of them.  The number of SNPs in overexpressed fragments appears to be 
clustered between 106 and 110 SNPs.  
<p>
This notebook explores series that express some of those fragments.  There appear to be three
main sources for the 123_1561 fragments.  One was the recombination event of series 8_267 in the
South Asian tree and the selection of that recombinant by the appearance of the series 9_39.  Another
was the hierarchy selected by the series 290_16.  The other was a variety of remodeling events as
parts of the 123_1561 SNPs were lost in the hierarchies associated with the series 28_434.
</div>

In [2]:
HTML(dm.series_html(dm.di_123_1561))

<h3>Series  123_1561 SNPs</h3>
<div style="width:700px">
<p>
The plot below shows the number of SNPs and the count of chromosome samples that express each number of
123_1561 SNPs.  The count data is also provided in the array below the plot.
</div>

In [3]:
count_data = an.sa_123_1561.unique_snps_per_allele()
plt_obj = smcp(count_data)
plt = plt_obj.do_plot()
show(plt)

In [4]:
count_data

array([(0, 2948), (1, 111), (2, 100), (3, 41), (4, 5), (5, 4), (12, 1),
       (17, 6), (18, 1), (23, 1), (24, 1), (25, 2), (51, 1), (52, 2),
       (66, 1), (86, 1), (87, 3), (88, 1), (90, 1), (91, 5), (95, 2),
       (96, 14), (105, 1), (106, 17), (107, 39), (108, 60), (109, 59),
       (110, 19), (113, 3), (115, 24), (116, 3), (117, 11), (118, 1),
       (119, 3), (120, 13), (121, 35), (122, 282), (123, 1186)], 
      dtype=[('count', '<u2'), ('snps', '<u2')])

<h3>Series 123_1561 fragments</h3>
<div style="width:700px">
<p>
The next several table cells show chomosome samples per SNP for 
the main fragments revealed in the 123_1561 count data.  The
count of chromosome samples that express an SNP is given for each SNP
in series 123_1561
</div>

In [5]:
aps_115, am_115 = an.sa_123_1561.snps_from_aps_value(115)
aps_115

array([24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24,
        0, 24, 24,  0, 24, 24,  0, 24, 24, 24, 24, 24, 24, 24, 24,  0, 24,
       24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24,
       24, 24, 24, 24, 24, 24, 24, 24, 24, 24,  0, 24, 24, 24, 24, 24, 24,
       24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24,  0, 24, 24, 24, 24,  0,
       24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24,
       24, 24, 24,  0, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24,
       24, 24, 24, 24])

In [6]:
aps_110, am_110 = an.sa_123_1561.snps_from_aps_value(110)
aps_110

array([19, 19, 19, 19,  1, 18,  1, 19, 19,  2,  3, 19, 19, 19,  1, 17,  5,
        1, 19, 19,  1, 19, 19,  1, 19, 19, 19, 19, 19, 19, 19, 19,  1, 19,
       19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19,
       19, 19, 19, 19, 19, 19, 19, 19, 19, 19,  1, 19, 19, 19, 19, 19, 19,
       19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 15, 19, 19, 19, 19,  1,
       19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19,
       19, 19, 19,  1, 19, 19, 19, 19, 18, 18, 18, 18, 18, 18, 18, 18, 18,
       18, 18, 18, 18])

In [7]:
aps_109, am_109 = an.sa_123_1561.snps_from_aps_value(109)
aps_109

array([59, 59, 59, 59,  0, 59,  0, 59, 59,  0,  0, 59, 59, 59,  0,  1,  0,
        0, 59, 59,  0, 59, 59,  0, 59, 59, 59, 59, 59, 59, 59, 59,  0, 59,
       59, 59, 59, 59, 59, 59, 59, 59, 59, 59, 59, 59, 59, 59, 59, 59, 59,
       59, 59, 59, 59, 59, 59, 59, 59, 59, 59,  0, 59, 59, 59, 59, 59, 59,
       59, 59, 59, 59, 59, 59, 59, 59, 59, 59, 59, 58, 59, 59, 59, 59,  0,
       59, 59, 59, 59, 59, 59, 59, 59, 59, 59, 59, 59, 59, 59, 59, 59, 59,
       59, 59, 59,  0, 59, 59, 59, 59, 59, 59, 59, 59, 59, 59, 59, 59, 59,
       59, 59, 59, 59])

In [8]:
aps_108, am_108 = an.sa_123_1561.snps_from_aps_value(108)
aps_108

array([60, 60, 60, 60,  5,  9,  5, 60, 60,  5,  0, 60, 60, 60,  5,  5,  5,
        5, 60, 60,  5, 60, 60,  5, 60, 60, 55, 55, 60, 60, 60, 60,  0, 60,
       60, 60, 55, 60, 60, 60, 60, 59, 60, 60, 60, 60, 60, 60, 60, 55, 60,
       60, 60, 60, 60, 60, 60, 60, 60, 60, 60,  0, 60, 60, 60, 60, 60, 60,
       60, 60, 60, 55, 60, 60, 60, 55, 60, 55, 60, 57, 60, 60, 60, 60,  0,
       60, 60, 60, 60, 60, 60, 60, 60, 60, 60, 60, 60, 60, 60, 60, 55, 55,
       60, 60, 60,  0, 60, 60, 60, 60, 60, 60, 60, 60, 60, 60, 60, 60, 60,
       55, 60, 60, 60])

In [9]:
aps_107, am_107 = an.sa_123_1561.snps_from_aps_value(107)
aps_107

array([39, 39, 39, 39, 28, 30, 28, 39, 39, 28, 28, 39, 39, 39, 28, 28, 28,
       27, 39, 38, 27, 39, 39, 27, 39, 39, 39, 39, 39, 39, 39, 39, 27, 38,
       39, 39, 39, 39, 39, 39, 39, 39, 39, 39, 39, 39, 39, 39, 39, 39, 39,
       39, 39, 39, 39, 39, 39, 39, 39, 39, 39, 27, 39, 39, 39, 39, 39, 39,
       39, 39, 39, 39, 39, 39, 39, 39, 39, 39, 39, 30, 39, 39, 39, 38, 27,
       39, 38, 38, 39, 39, 39, 38, 39, 38, 38, 38, 38, 38, 38, 39, 39, 39,
       39, 39, 39, 27, 39, 12, 12, 11, 12, 12, 12, 12, 12, 12, 12, 12, 12,
       12, 12, 12, 12])

In [10]:
aps_106, am_106 = an.sa_123_1561.snps_from_aps_value(106)
aps_106

array([17, 17, 17, 17, 16, 16, 16, 17, 17, 16, 16, 17, 14, 17, 16,  6, 16,
       15, 17, 17, 15, 17, 17, 15, 17, 17, 17, 16, 17, 17, 17, 17, 15, 17,
       17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17,
       17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 15, 17, 17, 17, 17, 17, 17,
       17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 15, 17, 17, 17, 16, 15,
       17, 16, 16, 17, 17, 17, 16, 17, 16, 16, 16, 16, 16, 16, 17, 17, 17,
       17, 17, 17, 15, 17,  2,  2,  1,  2,  2,  2,  2,  2,  2,  2,  2,  2,
        2,  2,  2,  2])

In [11]:
aps_96, am_96 = an.sa_123_1561.snps_from_aps_value(96)
aps_96

array([14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14,
       14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14,
       14, 14, 14, 14,  0,  0,  0, 14,  0,  0, 14, 14, 14, 14,  0, 14,  0,
        0,  0,  0, 14,  0, 14,  0,  0,  0,  0, 14,  0, 14, 14,  0, 14,  0,
        0, 14, 14, 14, 14, 14, 14, 14,  0,  0, 14,  0,  0, 14,  0, 14, 14,
       14, 14,  0, 14,  0,  0, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14,
       14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14,
       14, 14, 14, 14])

<h3>115 SNP fragments</h3>
<div style="width:700px">
<p>
The 115 SNP 123_1561 fragment is expressed by 24 chromosome samples.  All have lost
the same 8 SNPs.  Those SNPs are distributed among the 115 SNPs that all 24 samples
express.  This fragment has been selected by the series 25_27 that appeared in an
28_434 hierarchy that included both 123_1561 and 117_1685.  Chromosome samples that 
express 25_27 express 23 of the 24 115 SNP fragments.  Two other 25_27 samples
express a 116 SNP fragment with one additional SNP. The last 2 samples have lost
enough 123_1561 SNPs to no longer be counted as expressing it.
</div>

In [12]:
plt_obj = dm.superset_allele_mask(am_115, min_match=0.5)
plt = plt_obj.do_plot()
am = plt_obj.plot_context.yes_allele_mask
show(plt)

In [13]:
HTML(plt_obj.get_html())

In [14]:
an.sa_123_1561.alleles_per_snp(am)

array([24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24,
        0, 24, 24,  0, 24, 24,  0, 24, 24, 24, 24, 24, 24, 24, 24,  0, 24,
       24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24,
       24, 24, 24, 24, 24, 24, 24, 24, 24, 24,  0, 24, 24, 24, 24, 24, 24,
       24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24,  0, 24, 24, 24, 24,  0,
       24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24,
       24, 24, 24,  0, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24,
       24, 24, 24, 24])

In [15]:
di_25_27 = 353496
plt_obj = dm.superset_yes_no([di_25_27, dm.di_123_1561], min_match=0.5)
plt = plt_obj.do_plot()
am = plt_obj.plot_context.yes_allele_mask
show(plt)

In [16]:
HTML(plt_obj.get_html())

In [17]:
an.sa_123_1561.alleles_per_snp(am)

array([25, 25, 25, 25, 25, 25, 25, 25, 25, 25, 25, 25, 25, 25, 25, 25, 25,
        0, 25, 25,  0, 25, 25,  0, 25, 25, 25, 25, 25, 25, 25, 25,  0, 25,
       25, 25, 25, 25, 25, 25, 25, 25, 25, 25, 25, 25, 25, 25, 25, 25, 25,
       25, 25, 25, 25, 25, 25, 25, 25, 25, 25,  0, 25, 25, 25, 25, 25, 25,
       25, 25, 25, 25, 25, 25, 25, 25, 25, 25, 25,  2, 25, 25, 25, 25,  0,
       25, 25, 25, 25, 25, 25, 25, 25, 25, 25, 25, 25, 25, 25, 25, 25, 25,
       25, 25, 25,  0, 25, 25, 25, 25, 25, 25, 25, 25, 25, 25, 25, 25, 25,
       25, 25, 25, 25])

In [18]:
plt_obj = dm.superset_yes_no([di_25_27], [dm.di_123_1561], min_match=0.5)
plt = plt_obj.do_plot()
am = plt_obj.plot_context.yes_allele_mask
show(plt)

In [19]:
HTML(plt_obj.get_html())

In [20]:
an.sa_123_1561.alleles_per_snp(am)

array([2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 0, 2, 2, 0, 2, 2,
       0, 2, 2, 2, 2, 2, 2, 2, 2, 0, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2,
       2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 0, 2, 2, 2, 2, 2, 2, 2,
       2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 1, 2, 2, 2, 0, 0, 2, 0, 0, 2, 2, 2, 0,
       2, 0, 0, 0, 0, 0, 2, 2, 2, 2, 2, 2, 2, 0, 2, 2, 2, 2, 2, 2, 2, 2, 2,
       2, 2, 2, 2, 2, 2, 2, 2])

In [21]:
di_7_34 = 353563
plt_obj = dm.superset_yes_no([di_7_34, dm.di_24_1504], [di_25_27], min_match=0.5)
plt = plt_obj.do_plot()
am = plt_obj.plot_context.yes_allele_mask
show(plt)

In [22]:
HTML(plt_obj.get_html())

In [23]:
an.sa_123_1561.alleles_per_snp(am)

array([6, 6, 6, 6, 0, 5, 0, 6, 6, 0, 0, 6, 6, 6, 0, 4, 4, 0, 6, 6, 0, 6, 6,
       0, 6, 6, 6, 6, 6, 6, 6, 6, 0, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6,
       6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 0, 6, 6, 6, 6, 6, 6, 6,
       6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 0, 6, 6, 6, 6, 0, 6, 6, 6, 6, 6, 6, 6,
       6, 6, 6, 6, 6, 6, 5, 6, 6, 6, 6, 6, 6, 0, 6, 6, 6, 6, 6, 6, 6, 6, 6,
       6, 6, 6, 6, 6, 6, 6, 6])

<h3>110 SNP fragments</h3>
<div style="width:700px">
<p>
The 19 chromosome samples that express 110 SNP fragments include 17 that express
the 28_434 hierarchy selected by 6_68.
</div>

In [24]:
plt_obj = dm.superset_allele_mask(am_110, min_match=0.5)
plt = plt_obj.do_plot()
am = plt_obj.plot_context.yes_allele_mask
show(plt)

In [25]:
HTML(plt_obj.get_html())

In [26]:
an.sa_123_1561.alleles_per_snp(am)

array([19, 19, 19, 19,  1, 18,  1, 19, 19,  2,  3, 19, 19, 19,  1, 17,  5,
        1, 19, 19,  1, 19, 19,  1, 19, 19, 19, 19, 19, 19, 19, 19,  1, 19,
       19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19,
       19, 19, 19, 19, 19, 19, 19, 19, 19, 19,  1, 19, 19, 19, 19, 19, 19,
       19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 15, 19, 19, 19, 19,  1,
       19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19,
       19, 19, 19,  1, 19, 19, 19, 19, 18, 18, 18, 18, 18, 18, 18, 18, 18,
       18, 18, 18, 18])

<h3>109 SNP fragments</h3>
<div style="width:700px">
<p>
All 59 of the chromosome samples that express 109 SNP 123_1561 fragments
express the series 28_434.  They include 44 samples that express the series 6_68
</div>

In [27]:
plt_obj = dm.superset_allele_mask(am_109, min_match=0.5)
plt = plt_obj.do_plot()
am = plt_obj.plot_context.yes_allele_mask
show(plt)

In [28]:
HTML(plt_obj.get_html())

In [29]:
an.sa_123_1561.alleles_per_snp(am)

array([59, 59, 59, 59,  0, 59,  0, 59, 59,  0,  0, 59, 59, 59,  0,  1,  0,
        0, 59, 59,  0, 59, 59,  0, 59, 59, 59, 59, 59, 59, 59, 59,  0, 59,
       59, 59, 59, 59, 59, 59, 59, 59, 59, 59, 59, 59, 59, 59, 59, 59, 59,
       59, 59, 59, 59, 59, 59, 59, 59, 59, 59,  0, 59, 59, 59, 59, 59, 59,
       59, 59, 59, 59, 59, 59, 59, 59, 59, 59, 59, 58, 59, 59, 59, 59,  0,
       59, 59, 59, 59, 59, 59, 59, 59, 59, 59, 59, 59, 59, 59, 59, 59, 59,
       59, 59, 59,  0, 59, 59, 59, 59, 59, 59, 59, 59, 59, 59, 59, 59, 59,
       59, 59, 59, 59])

<h3>108 SNP fragments</h3>
<div style="width:700px">
<p>
The 60 chromosome samples that express 108 SNP fragments can be divided into three groups.
Two of them including 55 samples express 28_434.  One of those two is selected by the chromosome
samples that express 108 SNPs and the series 28_434 but not the series 10_17 or 4_21.  That group
includes 54 chromosome samples.  50 of those samples express the series 13_1696 and 22_73.  The
other 1 sample that expresses 28_434 expresses the series 10_17 and 6_68.
<p>
The remaining 5 chromosome samples express 108 123_1561 SNPs but do not express 28_434.  All
of those samples do express the series 4_21 and 13_1696 but not 117_1685.
</div>

In [30]:
di_10_17 = 353401
di_4_21 = 353388
plt_obj = dm.superset_allele_mask(am_108, [dm.di_28_434], [di_10_17, di_4_21], min_match=0.5)
plt = plt_obj.do_plot()
am = plt_obj.plot_context.yes_allele_mask
show(plt)

In [31]:
HTML(plt_obj.get_html())

In [32]:
an.sa_123_1561.alleles_per_snp(am)

array([54, 54, 54, 54,  0,  3,  0, 54, 54,  0,  0, 54, 54, 54,  0,  0,  0,
        0, 54, 54,  0, 54, 54,  0, 54, 54, 54, 54, 54, 54, 54, 54,  0, 54,
       54, 54, 54, 54, 54, 54, 54, 54, 54, 54, 54, 54, 54, 54, 54, 54, 54,
       54, 54, 54, 54, 54, 54, 54, 54, 54, 54,  0, 54, 54, 54, 54, 54, 54,
       54, 54, 54, 54, 54, 54, 54, 54, 54, 54, 54, 51, 54, 54, 54, 54,  0,
       54, 54, 54, 54, 54, 54, 54, 54, 54, 54, 54, 54, 54, 54, 54, 54, 54,
       54, 54, 54,  0, 54, 54, 54, 54, 54, 54, 54, 54, 54, 54, 54, 54, 54,
       54, 54, 54, 54])

In [33]:
plt_obj = dm.superset_allele_mask(am_108, [di_4_21], [di_10_17], min_match=0.5)
plt = plt_obj.do_plot()
am = plt_obj.plot_context.yes_allele_mask
show(plt)

In [34]:
HTML(plt_obj.get_html())

In [35]:
an.sa_123_1561.alleles_per_snp(am)

array([5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 0, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5,
       5, 5, 5, 0, 0, 5, 5, 5, 5, 0, 5, 5, 5, 0, 5, 5, 5, 5, 5, 5, 5, 5, 5,
       5, 5, 5, 0, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 0, 5, 5, 5, 5, 5, 5, 5,
       5, 5, 0, 5, 5, 5, 0, 5, 0, 5, 5, 5, 5, 5, 5, 0, 5, 5, 5, 5, 5, 5, 5,
       5, 5, 5, 5, 5, 5, 5, 5, 0, 0, 5, 5, 5, 0, 5, 5, 5, 5, 5, 5, 5, 5, 5,
       5, 5, 5, 5, 0, 5, 5, 5])

In [36]:
plt_obj = dm.superset_allele_mask(am_108, [dm.di_28_434, di_10_17], min_match=0.5)
plt = plt_obj.do_plot()
am = plt_obj.plot_context.yes_allele_mask
show(plt)

In [37]:
HTML(plt_obj.get_html())

In [38]:
an.sa_123_1561.alleles_per_snp(am)

array([1, 1, 1, 1, 0, 1, 0, 1, 1, 0, 0, 1, 1, 1, 0, 0, 0, 0, 1, 1, 0, 1, 1,
       0, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1])

<h3>107 SNP fragments</h3>
<div style="width:700px">
<p>
The 39 chromosome samples that express 107 SNP 123_1561 fragments include 27
that resulted from the recombination event selected by the series 9_39.  Another
11 chromosome samples express very different 107 SNP fragments along with the
series 28_434 and 49_136.  Those 11 samples include 8 that express 13_1696 and 22_73.
The remaining 107 123_1561 fragment appears to be very different than either of the
others.  The chromosome sample that expresses it also expresses 28_434 but not 49_136.
</div>

In [39]:
plt_obj = dm.superset_allele_mask(am_107, [dm.di_117_1685, dm.di_4_815], min_match=0.5)
plt = plt_obj.do_plot()
am_107_4_815 = plt_obj.plot_context.yes_allele_mask
show(plt)

In [40]:
HTML(plt_obj.get_html())

In [41]:
an.sa_123_1561.alleles_per_snp(am_107_4_815)

array([27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27,
       27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27,
       27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27,
       27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27,
       27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27,
       27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27,
       27, 27, 27, 27, 27,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,
        0,  0,  0,  0])

In [42]:
plt_obj = dm.superset_allele_mask(am_107, [dm.di_117_1685, dm.di_49_136], [dm.di_8_267], min_match=0.5)
plt = plt_obj.do_plot()
am_107_49_136_not_4_815 = plt_obj.plot_context.yes_allele_mask
show(plt)

In [43]:
HTML(plt_obj.get_html())

In [44]:
an.sa_123_1561.alleles_per_snp(am_107_49_136_not_4_815)

array([11, 11, 11, 11,  0,  2,  0, 11, 11,  0,  0, 11, 11, 11,  0,  0,  0,
        0, 11, 10,  0, 11, 11,  0, 11, 11, 11, 11, 11, 11, 11, 11,  0, 10,
       11, 11, 11, 11, 11, 11, 11, 11, 11, 11, 11, 11, 11, 11, 11, 11, 11,
       11, 11, 11, 11, 11, 11, 11, 11, 11, 11,  0, 11, 11, 11, 11, 11, 11,
       11, 11, 11, 11, 11, 11, 11, 11, 11, 11, 11,  2, 11, 11, 11, 11,  0,
       11, 11, 11, 11, 11, 11, 11, 11, 11, 11, 11, 11, 11, 10, 11, 11, 11,
       11, 11, 11,  0, 11, 11, 11, 10, 11, 11, 11, 11, 11, 11, 11, 11, 11,
       11, 11, 11, 11])

In [45]:
plt_obj = dm.superset_allele_mask(am_107, [dm.di_117_1685], [dm.di_49_136, dm.di_8_267], min_match=0.5)
plt = plt_obj.do_plot()
am_107_not_49_136_4_815 = plt_obj.plot_context.yes_allele_mask
show(plt)

In [46]:
HTML(plt_obj.get_html())

In [47]:
an.sa_123_1561.alleles_per_snp(am_107_not_49_136_4_815)

array([1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 0, 1, 1,
       0, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 1, 0, 0, 1, 1, 1, 0,
       1, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1])

<h3>106 SNP fragments</h3>
<div style="width:700px">
<p>
The 17 chromosome samples that express 106 SNP 123_1561 fragments include the remaining
15 samples that express the series 9_39.  The fragments expressed by those samples have lost
the same 16 SNPs at the high end of 123_1561 as the samples that express 107 SNP fragments.
They have also lost 1 of 4 other SNPs.  The other 2 chromosome samples that express 106
SNP fragments express 28_434.  The 2 fragments are different.
</div>

In [48]:
plt_obj = dm.superset_allele_mask(am_106, [dm.di_8_267], min_match=0.5)
plt = plt_obj.do_plot()
am_106_8_267 = plt_obj.plot_context.yes_allele_mask
show(plt)

In [49]:
HTML(plt_obj.get_html())

In [50]:
an.sa_123_1561.alleles_per_snp(am_106_8_267)

array([15, 15, 15, 15, 15, 15, 15, 15, 15, 15, 15, 15, 12, 15, 15,  5, 15,
       15, 15, 15, 15, 15, 15, 15, 15, 15, 15, 14, 15, 15, 15, 15, 15, 15,
       15, 15, 15, 15, 15, 15, 15, 15, 15, 15, 15, 15, 15, 15, 15, 15, 15,
       15, 15, 15, 15, 15, 15, 15, 15, 15, 15, 15, 15, 15, 15, 15, 15, 15,
       15, 15, 15, 15, 15, 15, 15, 15, 15, 15, 15, 15, 15, 15, 15, 15, 15,
       15, 15, 15, 15, 15, 15, 15, 15, 15, 15, 15, 15, 15, 14, 15, 15, 15,
       15, 15, 15, 15, 15,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,
        0,  0,  0,  0])

In [51]:
plt_obj = dm.superset_allele_mask(am_106, [dm.di_117_1685, dm.di_49_136], [dm.di_8_267], min_match=0.5)
plt = plt_obj.do_plot()
am_106_117_1685_49_136_not_8_267 = plt_obj.plot_context.yes_allele_mask
show(plt)

In [52]:
HTML(plt_obj.get_html())

In [53]:
an.sa_123_1561.alleles_per_snp(am_106_117_1685_49_136_not_8_267)

array([1, 1, 1, 1, 0, 0, 0, 1, 1, 0, 0, 1, 1, 1, 0, 0, 0, 0, 1, 1, 0, 1, 1,
       0, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 0, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1])

In [54]:
plt_obj = dm.superset_allele_mask(am_106, [dm.di_117_1685], [dm.di_49_136, dm.di_8_267], min_match=0.5)
plt = plt_obj.do_plot()
am_106_117_1685_not_49_136_8_267 = plt_obj.plot_context.yes_allele_mask
show(plt)

In [55]:
HTML(plt_obj.get_html())

In [56]:
an.sa_123_1561.alleles_per_snp(am_106_117_1685_not_49_136_8_267)

array([1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 0, 1, 1,
       0, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 0, 0, 1, 0, 0, 1, 1, 1, 0,
       1, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1])

<h3>96 SNP fragments</h3>
<div style="width:700px">
<p>
The 96 SNP 123_1561 SNP fragment was selected by the appearance of the 290 SNP series
290_16.  All 14 chromosome samples that express this fragment express 290_16.  The
other 2 samples that express 290_16 express a 95 SNP 123_1561 fragment that has
lost 1 of 2 SNPs that are part of the 96 SNP fragment.
</div>

In [57]:
plt_obj = dm.superset_allele_mask(am_96, min_match=0.5)
plt = plt_obj.do_plot()
am = plt_obj.plot_context.yes_allele_mask
show(plt)

In [58]:
HTML(plt_obj.get_html())

In [59]:
an.sa_123_1561.alleles_per_snp(am)

array([14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14,
       14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14,
       14, 14, 14, 14,  0,  0,  0, 14,  0,  0, 14, 14, 14, 14,  0, 14,  0,
        0,  0,  0, 14,  0, 14,  0,  0,  0,  0, 14,  0, 14, 14,  0, 14,  0,
        0, 14, 14, 14, 14, 14, 14, 14,  0,  0, 14,  0,  0, 14,  0, 14, 14,
       14, 14,  0, 14,  0,  0, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14,
       14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14,
       14, 14, 14, 14])

In [60]:
plt_obj = dm.superset_yes_no([dm.di_290_16], min_match=0.5)
plt = plt_obj.do_plot()
am = plt_obj.plot_context.yes_allele_mask
show(plt)

In [61]:
HTML(plt_obj.get_html())

In [62]:
an.sa_123_1561.alleles_per_snp(am)

array([16, 16, 16, 16, 16, 16, 16, 16, 16, 16, 16, 16, 15, 16, 16, 16, 16,
       16, 16, 16, 16, 16, 16, 16, 16, 16, 16, 16, 16, 16, 16, 16, 16, 16,
       16, 16, 16, 16,  0,  0,  0, 16,  0,  0, 16, 16, 16, 16,  0, 16,  0,
        0,  0,  0, 16,  0, 16,  0,  0,  0,  0, 16,  0, 16, 16,  0, 16,  0,
        0, 16, 16, 16, 16, 16, 16, 16,  0,  0, 15,  0,  0, 16,  0, 16, 16,
       16, 16,  0, 16,  0,  0, 16, 16, 16, 16, 16, 16, 16, 16, 16, 16, 16,
       16, 16, 16, 16, 16, 16, 16, 16, 16, 16, 16, 16, 16, 16, 16, 16, 16,
       16, 16, 16, 16])